framework/src/main/java/com/github/dbmdz/flusswerk/framework/rabbitmq/RabbitClient.java

```diff
@@ -27,19 +37,48 @@ public class RabbitClient {
 
   private static final boolean SINGLE_MESSAGE = false;
 
-  private Channel channel;
+  private static final Logger log = LoggerFactory.getLogger(RabbitClient.class);
 
-  private final FlusswerkObjectMapper objectMapper;
+  private final Channel channel;
+
+  private final Lock channelLock = new ReentrantLock();
+  private final Condition channelAvailableAgain = channelLock.newCondition();
+  private boolean channelAvailable = true;
```

Comment:
Do we need this flag in addition to the Condition `channelAvailableAgain`?

Reply:
Yes, they are for two separate scenarios, the signal only tells you when the channel is available *again*, but to be able to tell if the channel is available *now*, outside of a recovery scenario, we need the flag.